storage/cache/subtree_cache.go

```diff
@@ -46,8 +46,10 @@ type SubtreeCache struct {
 	dirtyPrefixes map[string]bool
 	// mutex guards access to the maps above.
 	mutex *sync.RWMutex
-
+	// used to rebuild internal nodes when subtrees are loaded
 	populateSubtree storage.PopulateSubtreeFunc
+	// used for preparation work when subtrees are about to be written to storage
+	prepareSubtreeWrite storage.PrepareSubtreeWriteFunc
```

Comment:
populate and prepare? I don't think it harms clarity because a.) there already isn't any and b.) they only get called once each.